Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,382 @@
+    f.edge = g.edge → f = g := OneTruncation₂.Hom.ext
+
+section
+variable {C : Type u} [Category.{v} C]
+
+/-- An equivalence between the type of objects underlying a category and the type of 0-simplices in
+the 2-truncated nerve. -/
+@[simps]
+def OneTruncation₂.nerveEquiv :
+    C ≃ OneTruncation₂ ((SSet.truncation 2).obj (nerve C)) where
+  toFun X := .mk₀ X
+  invFun X := X.obj' 0
+  left_inv _ := rfl
+  right_inv _ := ComposableArrows.ext₀ rfl
+
+/-- A hom equivalence over the function `OneTruncation₂.nerveEquiv.toFun`. -/
+def OneTruncation₂.nerveHomEquiv {X Y : C} : (X ⟶ Y) ≃ (nerveEquiv X ⟶ nerveEquiv Y) where
+  toFun f :=
+    { edge := ComposableArrows.mk₁ f
+      src_eq := ComposableArrows.ext₀ rfl
+      tgt_eq := ComposableArrows.ext₀ rfl }
+  invFun φ := eqToHom (congr_arg ComposableArrows.left φ.src_eq.symm) ≫ φ.edge.hom ≫
+      eqToHom (congr_arg ComposableArrows.left φ.tgt_eq)
+  left_inv f := by dsimp; simp only [comp_id, id_comp]; rfl
+  right_inv φ := by
+    ext
+    exact ComposableArrows.ext₁ (congr_arg ComposableArrows.left φ.src_eq).symm
+      (congr_arg ComposableArrows.left φ.tgt_eq).symm rfl
+
+/-- A hom equivalence over the function `OneTruncation₂.nerveEquiv.invFun`. -/
+def OneTruncation₂.nerveHomEquivInv {X Y : OneTruncation₂ ((SSet.truncation 2).obj (nerve C))} :
+  (X ⟶ Y) ≃ (nerveEquiv.invFun X ⟶ nerveEquiv.invFun Y) where
+  toFun φ := eqToHom (congr_arg ComposableArrows.left φ.src_eq.symm) ≫ φ.edge.hom ≫
+      eqToHom (congr_arg ComposableArrows.left φ.tgt_eq)
+  invFun f :=
+    { edge := ComposableArrows.mk₁ f
+      src_eq := ComposableArrows.ext₀ rfl
+      tgt_eq := ComposableArrows.ext₀ rfl }
+  left_inv φ := by
+    ext
+    exact ComposableArrows.ext₁ (congr_arg ComposableArrows.left φ.src_eq).symm
+      (congr_arg ComposableArrows.left φ.tgt_eq).symm rfl
+  right_inv f := by dsimp; simp only [comp_id, id_comp]; rfl
+
+/-- The refl prefunctor from the refl quiver underlying a nerve to the refl quiver underlying a
+category. -/
+def OneTruncation₂.ofNerve₂.hom : OneTruncation₂ ((SSet.truncation 2).obj (nerve C)) ⥤rq C where
+  obj := nerveEquiv.invFun
+  map := nerveHomEquivInv
+  map_id := fun X : ComposableArrows _ 0 => by
+    obtain ⟨x, rfl⟩ := X.mk₀_surjective
+    simp [nerveHomEquivInv]; rfl
+
+/-- The refl prefunctor from the refl quiver underlying a category to the refl quiver underlying
+a nerve. -/
+def OneTruncation₂.ofNerve₂.inv : C ⥤rq OneTruncation₂ (nerveFunctor₂.obj (Cat.of C)) where
+  obj := (.mk₀ ·)
+  map := fun f => by
+    refine ⟨.mk₁ f, ?_, ?_⟩ <;> apply ComposableArrows.ext₀ <;> simp <;> rfl
+  map_id _ := by ext; apply ComposableArrows.ext₁ <;> simp <;> rfl
+
+/-- The refl quiver underlying a nerve is isomorphic to the refl quiver underlying the category. -/
+def OneTruncation₂.ofNerve₂ (C : Type u) [Category.{u} C] :
+    ReflQuiv.of (OneTruncation₂ (nerveFunctor₂.obj (Cat.of C))) ≅ ReflQuiv.of C := by
```

Comment:
If would be good to have a constructor for isomorphisms in the category of quivers (resp. refl. quivers) which would take inputs like `nerveEquiv` and `nerveHomEquiv`.